app/views/directives/traffic-table.html

```diff
@@ -0,0 +1,84 @@
+<!-- Table combining routes and ports for a service -->
+<table class="table table-bordered table-hover table-mobile" style="{{ hasFooter ? 'margin-bottom:5px;' : ''}}">
+<thead>
+  <tr>
+    <th ng-if="!showNodePorts">{{customNameHeader || 'Route'}}</th>
+    <th ng-if="showNodePorts">{{customNameHeader || 'Route'}} / Node Port</th>
+    <th role="presentation"></th>
+    <th>Service Port</th>
+    <th role="presentation"></th>
+    <th>Target Port</th>
+    <th>Hostname</th>
+    <th>TLS Termination</th>
+  </tr>
+</thead>
+<tbody ng-if="(routes | hashSize) == 0 && (ports | hashSize) == 0">
+  <tr><td colspan="7"><em>{{emptyMessage || 'No routes or ports to show'}}</em></td></tr>
+</tbody>
+<tbody ng-if="(routes | hashSize) > 0">
```

Comment:
the route / port iteration and association logic in here isn't right, lets sync up later